core/src/main.cpp

```diff
@@ -20,7 +23,18 @@
 int main(int argc, char* argv[])
 {
 #ifdef USE_MPI
```

Comment:
The configuration parser is invoked by all the processes but does not perform any parallel communication and, therefore, does not need any knowledge of the MPI communicator.  
Would it be possible to postpone the MPI initialisation after the Configuration parsing so that the choice between `MPI_Init` and `oasis_c_init_comp` is not only depending on `USE_OASIS` but also on the activation of modules triggering OASIS ?

Reply:
I like this, as it also makes for cleaner code, in my opinion. See [d266c17](https://github.com/nextsimhub/nextsimdg/pull/641/commits/d266c17fa582f0635fd609a36244149be25364ef). I'm not sure if or how we can make the configuration decide if oasis is used, but the option is now there.